src/memory/memory_shared.h

```diff
@@ -102,4 +102,8 @@ void memory_read_shared_bootdata(chunk_shared_t* chunk_out);
  */
 USE_RESULT uint8_t memory_get_screen_type(void);
 
+#define MEMORY_SECURECHIP_TYPE_ATECC 0xFF
+#define MEMORY_SECURECHIP_TYPE_OPTIGA 0x01
+USE_RESULT uint8_t memory_get_securechip_type(void);
```

Comment:
@NickeZ why is this in memory_shared and not in memory? shared means shared with the bootloader, but the bootloader does not interact with the chip.

Reply:
If I may guess I wasn't aware at the time of the difference between the `memory.h` and the `memory_shared.h`. In my defence, `memory_shared.h` doesn't have a file-level comment specifying that it is only for things accessed from both firmware and bootloader.

Reply:
The struct you modified has that comment :stuck_out_tongue: 

https://github.com/BitBoxSwiss/bitbox02-firmware/blob/8cf1e44c0d1c849d1f617f9d7cc8e5f7ab510df6/src/memory/memory_shared.h#L61

But anyway, I realized that it has to be here anyway, as the bootloader will have to be able to report the securechip model to the host, so we can keep the option open to split the firmware release by securechip model if we run out of binary space.

Reply:
That isn't a file level comment. See this for example: https://github.com/torvalds/linux/blob/master/include/linux/list.h